iree_tests/update_config_xfails.py

```diff
@@ -83,27 +60,67 @@ def check_field(field_name):
         check_field("iree_compile_flags")
         check_field("iree_run_module_flags")
 
+    with open(args.log_file, "r") as f:
+        log_lines = f.readlines()
+
+    compile_failures = []
+    run_failures = []
+
+    for log_line in log_lines:
+        log_json = pyjson5.decode(log_line)
+
+        # Only look at test "call" events in the log.
+        if "when" not in log_json or log_json["when"] != "call":
+            continue
+
+        if log_json["outcome"] == "passed":
+            continue
+
+        # conftest.py adds extra info with `self.user_properties.append` for us
+        # to use. If that info is missing skip this log line.
+        if "user_properties" not in log_json or not log_json["user_properties"]:
+            logger.warning("Missing 'user_properties', ignoring log line")
+            continue
+        user_properties = log_json["user_properties"]
+
+        # TODO(scotttodd): handle multiple config files writing to one log file?
+
+        # Find the test directory name, since that is what our config.json uses
```

Comment:
When is the test directory name missing from the user properties? Is this something we need to find an alternative for?

Reply:
Looks like all of them have a node id if we can pull the test name from that

Reply:
Just defensive coding, in case this script gets out of sync with the log format / conftest.py. (I might copy this script into the IREE repo to make it easier to run without also cloning the test suite repo)

Reply:
Ok cool

Reply:
> Looks like all of them have a node id if we can pull the test name from that

I considered that, but wanted to use something that directly corresponds instead of parsing through e.g. `onnx/node/generated/test_cast_FLOAT_to_FLOAT16/model.mlir::cpu_llvm_sync_test` with regex.

I'm still not fully satisfied with how "test names" are modeled, so might do further refactoring there in the future.